src/frontend/src/players/pipelines/rtpUnpacker.ts

```diff
@@ -0,0 +1,73 @@
+import { Transform, TransformCallback, TransformOptions } from "stream"
+
+export class RtpPacket {
+  constructor(
+    public payloadType: number,
+    public sequenceNumber: number,
+    public timestamp: number,
+    public ssrc: number,
+    public payload: Buffer
+  ) {}
+}
+
+
+
+/**
+ * This class implements a Transform stream. 
+ * It is meant to be used as a Rtp packet payload unpacker for MPEGTS (MPEG Transport Stream specified by MPEG-2 part 1). 
+ * Currently, it assumes that each chunk represent one UDP packet of OZI Transport Layer that embed one whole RTP packet of OZI Application Layer.
+ */
+export class RtpUnpacker extends Transform {
+  constructor(options: TransformOptions = {}) {
+    super(options)
+  }
+
+  _transform(chunk: Buffer, _encoding: string, callback: TransformCallback): void {
+    const rtpHeader = chunk.slice(0, 12)
+    const hasPadding = Boolean(rtpHeader[0] & 0x20)
+    const marker = (rtpHeader[1]) >>> 7;
+    const payloadType = rtpHeader[1] & 0x7f
+
+    //  If the payload does not correspond to
+    //    1. A full <=> Marker = M = 1 
+    //    2. Audio/Video MPEG2 packet <=> Payload Type = PT = 33
+    //  the packet is left unchanged and passed through for FFmpeg to deal with.
+    //  Ref: https://en.wikipedia.org/wiki/Real-time_Transport_Protocol#:~:text=The%20RTP%20header%20has%20a,the%20particular%20class%20of%20application.
+    if(payloadType !== 33 || marker !== 1){
+      callback(null, chunk)
+    } else {
+      const sequenceNumber = rtpHeader.readUInt16BE(2)
+      const timestamp = rtpHeader.readUInt32BE(4)
+      const ssrc = rtpHeader.readUInt32BE(8)
+
+      let payloadStart = 12
+      let extensionLength = 0
+      if (rtpHeader[0] & 0x10) {
+        // Has Extension
+        extensionLength = rtpHeader.readUInt16BE(12)
+        payloadStart += 4 + extensionLength
+      }
+
+      if (hasPadding) {
+        const paddingLength = chunk[chunk.length - 1]
```

Comment:
I am not sure I understand the need for this whole `if` block. If the goal is to be able to determine the length of the payload, then it should be enough to do something like
`const payloadEnd = hasPadding ? chunk.length - chunk[chunk.length - 1] : chunk.length`
We may add an additional check if we want to make sure the paddingLength is positive

Reply:
You are right. Plus, it would avoid chunk manipulation. 

Those manipulation would make sense if chunks were concatenated in a container Buffer.